### PR TITLE
[TACHYON-1515] Audit misuse of ClientContext in tests.

### PR DESCRIPTION
**Pull request by luoli523**
In the tests, many testcase use ClientContext and reset some properties to meet the specific case.
And sometimes forget to reset back and makes other cases fail. It is safe to reset the ClientContext config after the case finish. 

And also, the ClientContext.reset() only reset the context in the first time, following call will just do nothing and return. To really reset the context , should call reset(conf)

### CODE REVIEW

**Comment by yupeng9** [comment]
LGTM. Thanks for the improvement!

**Comment by aaudiber** [comment]
Great catch @luoli523! I didn't realize the `reset()` method was useless for resetting the context. I think the problem is in the current `ClientContext.reset()` method - it should reset the context regardless of whether it's been set already. We should remove the check for
```
if (sInitialized) {
  return;
}
```
Then we can use `reset()` to actually reset the context.

File: integration-tests/src/test/java/tachyon/shell/TfsShellTest.java [hunk 1]
```diff
@@ -1223,5 +1223,6 @@ public void persistDirectoryTest() throws IOException, TachyonException {
     checkFilePersisted(mTfs.open(new TachyonURI("/testWildCards/foo/foobar2")), 20);
     checkFilePersisted(mTfs.open(new TachyonURI("/testWildCards/bar/foobar3")), 30);
     checkFilePersisted(mTfs.open(new TachyonURI("/testWildCards/foobar4")), 40);
+    ClientContext.getConf().merge(new TachyonConf());
```
Comment (aaudiber) [inline integration-tests/src/test/java/tachyon/shell/TfsShellTest.java]: should this be `reset` instead?

**Comment by luoli523** [comment]
@aaudiber  OK, I also think it's a better choice to remove the check below
```java
if (sInitialized) {
  return;
}
```

I will update the patch.

**Comment by luoli523** [comment]
updated the PR
Remove the reset check in ClientContext and call the reset() in the related TestCases.

@aaudiber  Please check this.

**Comment by gpang** [comment]
LGTM

**Comment by aaudiber** [comment]
+1 LGTM

**Pull request merged**